### PR TITLE
network,dhcp: fix theoretical race in device renaming

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/libsystemd-network/dhcp-identifier.c [hunk 1]
```diff
@@ -207,48 +203,20 @@ int dhcp_identifier_set_duid(
...
         const char *name = NULL;
         uint32_t id32;
         uint64_t id;
-        int r;
 
-        assert(ifindex > 0);
         assert(hw_addr);
         assert(ret);
 
-        if (udev_available() && !use_mac) {
```
Comment (keszybz) [inline src/libsystemd-network/dhcp-identifier.c:228]: Previously this was conditionalized on `use_mac`, which was (usually) set to `client->test_mode`. Now this is unconditional if `dev` is set. Can you comment a bit on the effect?

**Review by keszybz — COMMENTED** [review]
Is the race scenario described in the last commit probable? I.e. is this fixing a known bug…

The new code looks cleaner than the old one. I'm not exactly sure what all the implications of the change are though. If you could describe that in the commit message that'd be great.

**Comment by yuwata** [comment]
> Is the race scenario described in the last commit probable? I.e. is this fixing a known bug…

Yes, at least theoretically. But there is no report.
IIRC, I saw several CI failures caused by the code path, though it is not caused by double renaming.

The commit message of the last commit is updated. PTAL.

**Comment by yuwata** [comment]
(As I wrote in the update commit message, this is mostly cleanup, and only fixes an issue in super corner case. Hence not necessary to be in v252.)

**Review by bluca — APPROVED** [review]

**Pull request merged**